File: 2958/2958.py
class Solution:
    def length_of_longest_subarray_with_at_most_(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.length_of_longest_subarray_with_at_most_(root.left), self.length_of_longest_subarray_with_at_most_(root.right))

File: 2958/2958.cpp
class Solution {
public:
    int lengthOfLongestSubarrayWithAtM(TreeNode* root) {
        if(!root) return 0;
        return 1+max(lengthOfLongestSubarrayWithAtM(root->left),lengthOfLongestSubarrayWithAtM(root->right));
    }
};